sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -181,6 +203,10 @@ export class TestProxyHttpClient {
         const req = this._createRecordingRequest(stopUri);
         req.headers.set("x-recording-save", "true");
 
+        if (isRecordMode()) {
+          req.headers.set("Content-Type", "application/json");
+          req.body = JSON.stringify(variables ?? this.variables);
```

Comment:
Wouldn’t this mean that we’re setting `req.body` to undefined if there are no variables? Is that desired?

Reply:
`this.variables` is initialized as `{}`.
![image](https://user-images.githubusercontent.com/10452642/141029050-f001cda1-baca-4583-99f9-681880a0e895.png)

So, no problem.